### PR TITLE
Added functionality to append modal on click of node

### PR DESCRIPTION
**Pull request by ghost**
In this PR I: 
- Wrote a function that appends a modal on click of each node, passing in the data relevant to that particular node
- populated this modal with basic info (eg image/text etc. audio and video will be added in once we have sorted out S3.
- added a close button that removes all modal svg elements
- appended the heart svg to the modal. This will need to have an on click function added to it at a later stage so users can like their memories
- created a function that opens a random memory when the shuffle button is clicked
- updated the db_build.sql file so that our generic starting memories have a working image url/ text paragraph.

### CODE REVIEW

**Review by pbywater — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -31,9 +31,12 @@
     "tap-spec": "^4.1.1"
   },
   "dependencies": {
+    "are-we-there-yet": "^1.1.4",
```
Comment (des-des) [inline package.json]: Not using

File: package.json [hunk 2]
```diff
@@ -31,9 +31,12 @@
...
     "body-parser": "^1.17.2",
     "env2": "^2.2.0",
     "express": "^4.15.3",
-    "pg": "^6.4.0"
+    "pg": "^6.4.0",
+    "signal-exit": "^3.0.2",
```
Comment (des-des) [inline package.json]: Not using

File: package.json [hunk 3]
```diff
@@ -31,9 +31,12 @@
...
+    "wide-align": "^1.1.2"
```
Comment (des-des) [inline package.json]: Not using

File: public/assets/icons/card_icons/close.svg [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+<!-- Generator: Adobe Illustrator 19.0.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
+<svg version="1.1" id="Capa_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
+	 viewBox="0 0 47.971 47.971" style="enable-background:new 0 0 47.971 47.971;" xml:space="preserve">
+<g>
+	<path d="M28.228,23.986L47.092,5.122c1.172-1.171,1.172-3.071,0-4.242c-1.172-1.172-3.07-1.172-4.242,0L23.986,19.744L5.121,0.88
+		c-1.172-1.172-3.07-1.172-4.242,0c-1.172,1.171-1.172,3.071,0,4.242l18.865,18.864L0.879,42.85c-1.172,1.171-1.172,3.071,0,4.242
+		C1.465,47.677,2.233,47.97,3,47.97s1.535-0.293,2.121-0.879l18.865-18.864L42.85,47.091c0.586,0.586,1.354,0.879,2.121,0.879
+		s1.535-0.293,2.121-0.879c1.172-1.171,1.172-3.071,0-4.242L28.228,23.986z"/>
+</g>
+<g>
```
Comment (des-des) [inline public/assets/icons/card_icons/close.svg]: ?

File: public/assets/icons/card_icons/close.svg [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+</g>
+<g>
+</g>
+<g>
+</g>
+<g>
+</g>
+<g>
```
Comment (des-des) [inline public/assets/icons/card_icons/close.svg]: ?

**Review by des-des — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (pbywater) [reply package.json]: For some reason the server won't run without these and  the ones below!

**Comment by pbywater** [comment]
Also adds:

- queries for getting and updating likes in the database
- number of likes on like button
- updates number of likes when button is clicked
- updates number of likes in database when button is clicked

File: package.json (hunk 1 above)
Comment (pbywater) [reply package.json]: @des-des

File: package.json (hunk 3 above)
Comment (ConchobarMacNessa) [inline package.json]: We tried to get rid of all these unnecessary dependencies, but the server won't start without them.. we left them in our most recent PR..

File: public/main.css [hunk 6]
```diff
@@ -589,3 +589,26 @@ label.memory-input__label {
...
+    height: 40px;
+  }
+  100%: {
+    width: 40px;
+    height: 40px;
+  }
+}
+
+.like-button.liked {
+transition: 1s grow forwards;
```
Comment (des-des) [inline public/main.css]: Indentation

File: src/database/db_update.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
+const connect = require('./db_connect');
+
+const updateLikes = (likeNum, callback) => {
+  connect.query(
+    'UPDATE memories SET likes = $1', [likeNum], (err, res) => {
```
Comment (des-des) [inline src/database/db_update.js]: Just pass the callback straight into the query func

Comment (des-des) [reply src/database/db_update.js]:
```js
connect.query('UPDATE memories SET likes = $1', [likeNum], callback)
```

**Review by des-des — CHANGES_REQUESTED** [review]

File: src/database/db_update.js [hunk 8]
```diff
@@ -0,0 +1,15 @@
+const connect = require('./db_connect');
+
+const updateLikes = (likeNum, memoryId, callback) => {
+  connect.query(
+    'UPDATE memories SET likes = $1 WHERE memories.id = $2', [likeNum, memoryId], (err, res) => {
```
Comment (des-des) [inline src/database/db_update.js]:
```js
connect.query('QUERY', [...args], callback)
```
You do not need to define that cb, it does nothing

File: src/server/controllers/getLikes.js [hunk 9]
```diff
@@ -0,0 +1,15 @@
+const express = require('express');
+require('env2')('./config.env');
+const { getLikes } = require('./../../database/db_get');
+
+
+module.exports = (req, res) => {
+  const memoryId = req.params.memoryId;
+  getLikes(memoryId, (error, response) => {
+    if (error) {
+      if (error) console.log('getLikes error is', error);
```
Comment (des-des) [inline src/server/controllers/getLikes.js]:
```js
if (error) {
  console.error('getLikes error is', error);
  res.status(500).send(error)
  return ;
}

File: src/server/controllers/getLikes.js [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+    }
+    const num = response.toString();
```
Comment (des-des) [inline src/server/controllers/getLikes.js]: num is bad name

File: src/server/controllers/updateLikes.js [hunk 11]
```diff
@@ -0,0 +1,12 @@
+const express = require('express');
+require('env2')('./config.env');
+const { updateLikes } = require('./../../database/db_update');
+
+module.exports = (req, res) => {
+  updateLikes(req.body.numLikes, req.body.memoryId, (error, response) => {
+    if (error) {
+      if (error) console.log('update error is', error);
```
Comment (des-des) [inline src/server/controllers/updateLikes.js]: same as above

File: src/server/router.js [hunk 12]
```diff
@@ -13,5 +13,8 @@ router.post('/memory-input-text', require('./controllers/addTextMemory.js'));
 router.route('/memories')
   .get(require('./controllers/getMemories.js'))
   .delete(require('./controllers/deleteMemory.js'));
+router.route('/likes')
```
Comment (des-des) [inline src/server/router.js:17]: :+1:

**Review by des-des — CHANGES_REQUESTED** [review]

**Comment by des-des** [comment]
@pbywater not sure what is going on, but you need to push your fixes to this branch (responses to comments only)

**Comment by pbywater** [comment]
@des-des I'm confused! The fixes are above no?

**Comment by des-des** [comment]
These commits are somewhere else, 
image

**Comment by pbywater** [comment]
@des-des Just checked the branch - all the changes are on this branch!

File: src/database/db_update.js [hunk 13]
```diff
@@ -0,0 +1,15 @@
+const connect = require('./db_connect');
+
+const updateLikes = (likeNum, memoryId) => {
```
Comment (des-des) [inline src/database/db_update.js]: Okay, I do not think you understood me, this function will no longer work. You cannot return from an async function, it does not work!

Before you had
```js
const updateLikes = (arg, cb) => {
  query('SOME QUERY', [], (err, res) => {
    if (err) {
      cb(err);
      return;
    }
    cb(null, res)
  }
}
```

I was suggesting you changed it too

```js
const updateLikes = (arg, cb) => {
  query('SOME QUERY', [], cb)
}
```

As these two are identical

**Review by des-des — CHANGES_REQUESTED** [review]

**Comment by pbywater** [comment]
@des-des Like that?

**Comment by des-des** [comment]
@pbywater perf :ok_hand:

**Review by des-des — APPROVED** [review]

**Pull request merged**